languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml

```diff
@@ -2909,8 +2909,28 @@ USA
                 <token spacebefore="no">i</token>
             </antipattern>
             <rule>
+                <!-- antipatterns intended for mathematical or scientific usages of the symbol "i" -->
                 <antipattern>
-                    <token regexp="yes">row|index</token>
+                    <token regexp="yes">(row|index|variable|quantit(y|ie)|number|term|symbol|constant|point|element|unit|value|size|equal)s?|sub|negative|minus|lowercase</token>
+                    <token>i</token>
+                </antipattern>
+                <antipattern>
+                    <token>square</token>
+                    <token>of</token>
+                    <token>i</token>
+                </antipattern>
+                <antipattern>
+                    <token>equal</token>
+                    <token>to</token>
+                    <token>i</token>
```

Comment:
Same comment as above: does "to I" ever occur regularly? In a mathematical sense, it would appear more frequently, I guess:

_The distance from the origin to i is 1._
_In addition to i, quaternions also have j and k._